app/scripts/modules/amazon/src/function/configure/FunctionBasicInformation.tsx

```diff
@@ -0,0 +1,271 @@
+
+    const form = {
+      account$: formValues$.map(x => x.credentials).distinctUntilChanged(),
+      region$: formValues$.map(x => x.region).distinctUntilChanged(),
+      functionName$: formValues$.map(x => x.functionName).distinctUntilChanged(),
+      runtime$: formValues$.map(x => x.runtime).distinctUntilChanged(),
+      s3bucket$: formValues$.map(x => x.s3bucket).distinctUntilChanged(),
+      s3key$: formValues$.map(x => x.s3key).distinctUntilChanged(),
+      handler$: formValues$.map(x => x.handler).distinctUntilChanged(),
+    };
+
+    const allAccounts$ = Observable.fromPromise(AccountService.listAccounts('aws')).shareReplay(1);
+
+    // combineLatest with allAccounts to wait for accounts to load and be cached
+    const accountRegions$ = Observable.combineLatest(form.account$, allAccounts$)
+      .switchMap(([currentAccount, _allAccounts]) => AccountService.getRegionsForAccount(currentAccount))
+      .shareReplay(1);
+
+    const allFunctions$ = this.props.app.getDataSource('functions').data$ as Observable<IAmazonFunction[]>;
+    const regionfunctions$ = Observable.combineLatest(allFunctions$, form.account$, form.region$)
+      .map(([allFunctions, currentAccount, currentRegion]) => {
+        return allFunctions
+          .filter(fn => fn.account === currentAccount && fn.region === currentRegion)
+          .map(fn => fn.functionName);
+      })
+      .shareReplay(1);
+
+    accountRegions$
+      .withLatestFrom(form.region$)
+      .takeUntil(this.destroy$)
+      .subscribe(([accountRegions, selectedRegion]) => {
+        // If the selected region doesn't exist in the new list of regions (for a new acct), select the first region.
+        if (!accountRegions.some(x => x.name === selectedRegion)) {
+          this.props.formik.setFieldValue('region', accountRegions[0] && accountRegions[0].name);
+        }
+      });
+
+    Observable.combineLatest(allAccounts$, accountRegions$, regionfunctions$)
+      .takeUntil(this.destroy$)
+      .subscribe(([accounts, regions, existingFunctionNames]) => {
+        return this.setState({ accounts, regions, existingFunctionNames });
+      });
+  }
+
+  private handleRegionChange = (value: string): void => {
+    this.props.formik.setFieldValue('region', value);
+  };
+
+  public render() {
+    const { isNew } = this.props;
+    const { errors, values } = this.props.formik;
+    const { accounts, regions } = this.state;
+    const className = classNames({
+      well: true,
+      'alert-danger': errors.functionName,
+      'alert-info': !errors.functionName,
+    });
+    return (
+      <div className="container-fluid form-horizontal ">
+        <div className="sp-margin-m-bottom">
+          {isNew && (
+            <div className={className}>
+              <strong>Your load balancer will be named: </strong>
+              <HelpField id="aws.function.name" />
+              <span>{this.props.app.name.concat('-').concat(values.functionName)}</span>
+              <Field
```

Comment:
We're working towards consolidating forms and form error handling in Deck.  To that end, I'd like to try using a `FormikFormField` here.  If it doesn't work, then what you have already is fine.

```
<FormikFormField name="functionName" input={() => null} />
```

Reply:
Also, I notice that we have a FormikFormField on line 206 that should already be handling the error state.  Why are we displaying the same error here?  Can we delete this `<Field/>` and `<ValidationMessage/>`  completely?

Reply:
We can delete it. If we do so, it displays a red border indicating that there is an error. It does not display the error message.
I am deleting it and pushing it in the my next commit. :)